core/app/models/concerns/spree/product_scopes.rb

```diff
@@ -94,6 +94,19 @@ def self.property_conditions(property)
         taxons.first ? prepare_taxon_conditions(taxons) : where(nil)
       end
 
+      add_search_scope :ascend_by_taxons_min_position do |taxon_ids|
```

Comment:
@damianlegawiec @MatthewKennedy I'm wondering if it'd be worth putting that on the Product model as a counter column. This scope works rather fine, but it messes up using aggregate functions later.

Like here: https://github.com/spree/spree/pull/11047/files#diff-51600718114dcbd23bd049965e56224fcf19c430a61a38a1c0aa9e263aaece70R65